src/module.ts

```diff
@@ -148,12 +154,15 @@ export default defineNuxtModule<ModuleOptions>({
     addPlugin(resolve('./runtime/plugin'))
 
     // 7. Setup next-auth env-variables just to be safe, see https://github.com/nextauthjs/next-auth/blob/6280fe9e10bd123aeab576398d1e807a5ac37edc/apps/playground-nuxt/src/module.ts#L32-L38
-    extendViteConfig((config) => {
-      config.define = config.define || {}
-      config.define['process.env.NEXTAUTH_URL'] = JSON.stringify(url)
-      config.define['process.env.NEXTAUTH_URL_INTERNAL'] = JSON.stringify(url)
-      config.define['process.env.VERCEL_URL'] = JSON.stringify(process.env.VERCEL_URL)
-    })
+    if (!nuxt.options.dev && isOriginSet) {
+      extendViteConfig((config) => {
+        config.define = config.define || {}
+        config.define['process.env.NEXTAUTH_URL'] = JSON.stringify(url)
+        config.define['process.env.NEXTAUTH_URL_INTERNAL'] = JSON.stringify(url)
```

Comment:
out of interest: where you able to verify that `NextAuth` correctly picks these up? When I took over this code from the app-example I did not attempt to verify it's supposed functionality.

Reply:
Yeah, me neither... I don't know if it actually works...

Reply:
I searched for envs in NextAuth repo and it seems that the core is not using envs. So I think it's safe to remove them.

Reply:
What about: https://github.com/nextauthjs/next-auth/blob/7e91d7df54f9d8ca782c93a63f583ce89c2d178d/packages/next-auth/src/utils/node.ts#L37-L39

References:
![image](https://user-images.githubusercontent.com/25911230/205690010-865006a8-0adb-45b8-a4db-7b434abc91a3.png)

-> If based on this it remains unclear to us, I would leave this part as:
- it contains only information that can be public
- does not incur added runtime cost
- is also part of the nextauth nuxt example

Reply:
From what I see, this util is only used for `next` module. And as the next-auth wants to be agnostic, I believe they will not be using envs in core.

Reply:
:+1: very good, thanks for further looking into  it. Then let's drop it (: